### PR TITLE
Replace gophermail by jordan-wright/email

### LINKED ISSUE #10703: mail package deprecated
We have an immediate need to fix these issues with the way Sourcegraph sends email:

- https://github.com/sourcegraph/sourcegraph/issues/10702
- https://github.com/sourcegraph/sourcegraph/issues/10700

But the upstream mail sending package we use https://github.com/jpoehls/gophermail has been archived and unmaintained for 4 years. Also, none of the mail sending code in Sourcegraph is tested so swapping it out is going to be risky.

https://github.com/go-gomail/gomail looks like a promising package to move to. Once implemented, we should manually test all email user flows work:

- saved search "subscribe" emails
- saved search notification emails
- saved search "unsubscribe" emails
- organization invite emails
- reset password emails
- emails on sourcegraph.com which use mandrill which I think we have some special code to disable click tracking
- Every other place we call txemail.send that might not work due to a difference in underlying mail package behavior

### PR DESCRIPTION
**Pull request by asdine**
This PR fixes #10703 by replacing the unmaintainted [jpoehls/gophermail](https://github.com/jpoehls/gophermail) package by [jordan-wright/email](https://github.com/jordan-wright/email).

### CODE REVIEW

**Comment by asdine** [comment]
@unknwon Not yet, but I won't merge until it's done 👍

File: internal/txemail/template_test.go [hunk 1]
```diff
@@ -35,10 +35,10 @@ func TestParseTemplate(t *testing.T) {
...
-	if want := `a text body <b>`; m.Body != want {
-		t.Errorf("got text body %q, want %q", m.Body, want)
+	if want := `a text body <b>`; string(m.Text) != want {
+		t.Errorf("got text body %q, want %q", string(m.Text), want)
 	}
-	if want := `a html body <span class="&lt;b&gt;" />`; m.HTMLBody != want {
-		t.Errorf("got html body %q, want %q", m.HTMLBody, want)
+	if want := `a html body <span class="&lt;b&gt;" />`; string(m.HTML) != want {
+		t.Errorf("got html body %q, want %q", string(m.HTML), want)
 	}
```
Comment (keegancsmith) [inline internal/txemail/template_test.go:43]: minor: maybe cleaning up this test to use `cmp.Diff` would make it easier to read and less annoying to update :)

Comment (asdine) [reply internal/txemail/template_test.go:43]: Good idea :)

**Comment by asdine** [comment]
QA done with @unknwon 🎉

**Pull request merged**